### DISCUSSION TITLE
Configurable layout?

### DISCUSSION THREAD

**Discussion by raffraffraff** [quodlibet/quodlibet#3912] [Q&A] [answered]
I'd love to be able to modify the layout of quodlibet, but this doesn't seem to be possible within the app. Is it non-trivial to do modify the code? Or is there a nice shortcut like a layout.py or css file?  I'm cloning the project to look for myself, but if someone has already been here and can give me a quick pointer, that would be great

**Comment by jej** [comment]
I would say yes and no!
What layout are you thinking about?

**Reply by raffraffraff** [reply]
[edit: didn't add this as a reply at first, sorry]
I'm a MusicBee user, but I'm also a Linux user, and it sucks to get MusicBee working on Linux. I can't unilaterally move to QuodLibet because I'm not the only person in the house who uses it. Ideally, I want my current MusicBee layout:

[MusicBee screenshot](https://imgur.com/a/qIub5Mq)

Left pane: source selection (eg: library, playlist, file browser, soundcloud etc)
Middle pane: display whatever is selected in the left pane, with configurable columns and a search filter
Right pane: "Now Playing" (ie: the queue, in quodlibet parlance) with configurable columns and the play controls.

That's basically it. I don't care about album art or visualizations. I care about being able to quickly search / sort / filter my music collection using any of the tags I've added over 20+ years.

**Reply by jej** [reply]
> Left pane: source selection (eg: library, playlist, file browser, soundcloud etc)

Fun. Last week I was thinking about a way to add a left sidebar, like some players have... Plugin or not plugin, anyway I didn't reach easily to add a sidebar on the left in the current code.
I guess that it should be coded in a new browser: https://github.com/quodlibet/quodlibet/tree/master/quodlibet/browsers

**Reply by jej** [reply]
See also #2741

**Reply by raffraffraff** [reply]
As a complete hack, what would work for me short term is the "Browse > Playlists" view, _if_ the playlists view was in any way decent. But it's not. It's far too limited. It doesn't support any sort of structure (folders, subfolders). It doesn't seem to have smart playlists. If those features existed, I could just create a folder called "Playlists" that has my static playlists, another called "Smart Playlists", and at the top level I could create a single smart playlist called "Library" that shows all tracks (ie: match on some ubiquitous tag, like "Year > 0").

**Reply by raffraffraff** [reply]
Interesting, so does https://github.com/quodlibet/quodlibet/pull/2538 mean I can create my own sidebar plugins for this? Would a sidebar plugin be able to modify the content of the main window (the way it gets updated when you click on a playlist in the Browse > Playlists view)?

**Reply by jej** [reply]
I think the smartlists are the saved searches in QL: https://quodlibet.readthedocs.io/en/latest/guide/searching.html#searching

**Reply by raffraffraff** [reply]
Ah. Right. I enabled that, but I can't see any new UI features. It looks like like 'saved searches' aren't represented anywhere in the UI, they just show up as a drop-down history for the search field. Not good for me. I can't organize them, add them to the playlists view etc.

**Reply by jej** [reply]
> does #2538 mean I can create my own sidebar plugins

I think it's for the right sidebar.

**Reply by jej** [reply]
> It looks like like 'saved searches' aren't represented anywhere in the UI, they just show up as a drop-down history for the search field. Not good for me.

Yep, but I guess it's a starting point the make changes in the QL way... A left sidebar could display these saved searches, and profit of the underlayer to make the filtering.

**Reply by raffraffraff** [reply]
OK, so the sidebar plugins appear on the right. I just tried one, I see what you mean. I think that would work better for the "Now Playing" view, since I don't like where the queue is currently displayed at the bottom. Maybe I can create a simple plugin that just shows the existing queue view

As for the left pane, I think it might be easier to play around with an existing browser (eg: Album List, or File System) to see if I can come up with something usable.

**Reply by jej** [reply]
> Maybe I can create a simple plugin that just shows the existing queue view

I would go directly for a new browser with all these changes, a kind of "musicbee" browser. But I am not enough experienced with QL, maybe wait for more answers from @declension, @lazka, @bassstorm, etc...

**Comment by declension** [comment, marked as answer]
Definitely some good suggestions there already. 

@raffraffraff It sounds like if the queue could dock to the right instead of the bottom, this would solve a large part of the UI challenge?

Meanwhile, background reading:
For showing searches in the UI: #205
See also #152 for original discussions, #2042, #157, #3465 (maybe)
but especially #2273

**Comment by FoobyLinux** [comment]
Good evening everybody!

About 1,5 years ago, I've changed to Linux (for the 3rd or 4th time in my life) - seriously decided to stay this time.
One thing, that I've been struggling with in vain is to get foobar2000 to play music without resampling. As my latest attempt (Manjaro 25.0, Wine 10.5-1, Foobar 2.24.1) brought more problems than before, I turned to look more closely at music players under Linux.
Quod libet is able to read custom tags. In theory, it is able to handle smart or auto playlists. I managed relatively fast to stop it resampling the music files during playback.

But is there any way to display the album art permanently in a part of the GUI?
And can I make the search based auto playlists accessible easily?

In foobar it took me about half an hour after installation to have a look like in the attached screenshot. I'd be happy to reach a fairly comparable result in Quod libet.
![Bildschirmfoto_2025-04-16_21-35-20](https://github.com/user-attachments/assets/85ad33be-d7d6-4f6c-b13a-cbc1f833b16e)

Any help is appreciated.

Thank you very much in advance and best regards

Mat(thias)

**Reply by declension** [reply]
> But is there any way to display the album art permanently in a part of the GUI?

No, but see some of the linked issues. Should be relatively easy to to as, say, a sidebar plugin (but only as full-height, currently - layout doesn't support otherwise)

> And can I make the search based auto playlists accessible easily?

In _one_ window, no. But [secondary browsers](https://quodlibet.readthedocs.io/en/quodlibet-4.6/guide/browse/overview.html#secondary-browsers) are for that purpose, so some people open another (secondary) window to have playlists (or vice versa)

**Comment by FoobyLinux** [comment]
Thank you for the answer. Meanwhile, I've changed to JRiver MediaCenter. Sorry for not having withdrawn my question.
Quod libet does not seem to be able to read metadata from tagged WAV-files. That stopped me from finding a GUI that would have worked for me.

**Reply by declension** [reply]
See #3556

**Comment by Sappelen** [comment]
Yes please! I'd like to decrease the margins in the cover grid and get rid of the rounded edges of the images. Can anyone give me some pointers on how that could be achieved?

**Reply by declension** [reply]
Please raise a separate issue / discussion, too easy for this to get lost

**Reply by Sappelen** [reply]
Thx, will look into that later on
